### PR TITLE
add `enableSound` parameter to `dwarf-fortress` derivation

### PR DESCRIPTION
**Pull request by matthew-piziak**
###### Motivation for this change

The init file for this derivation can include `[SOUND:NO]` or `[SOUND:YES]`. This adds a parameter to control it. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/dwarf-fortress/wrapper/default.nix
Comment (Baughn) [inline pkgs/games/dwarf-fortress/wrapper/default.nix:72]: I do believe you need to call unBool here.

Comment (matthew-piziak) [reply pkgs/games/dwarf-fortress/wrapper/default.nix]: Oh, how embarrassing, I squashed my test commit into the contribution. Thank you @Baughn, I've fixed it.

**Review by Baughn — APPROVED** [review]

**Comment by matthew-piziak** [comment]
Thanks for the approval! I'm new to nixpkgs. How should I merge these in? Do I ask a bot or a maintainer, or what?

**Comment by Baughn** [comment]
We need someone with write access to merge it. To be honest, I'm not sure how to attract their attention from this side... let's give it a day and see what happens.

**Comment by adisbladis** [comment]
I haven't tested this myself, but since I see nothing wrong with the diff and it has approval by the package maintainer I'm merging.

**Pull request merged**
